src/provider/provider_tracking.c

```diff
@@ -230,40 +444,57 @@ static umf_result_t trackingAllocationSplit(void *hProvider, void *ptr,
         goto err;
     }
 
+    assert(level < MAX_LEVELS_OF_ALLOC_SEGMENT_MAP);
+    int cret =
+        critnib_insert(provider->hTracker->alloc_segments_map[level],
```

Comment:
Don't we have to increase the parent's n_children by 1 here as well?

Reply:
Yes, but not here - `critnib_insert()` only updates the existing entry here.
It should be increased in `umfMemoryTrackerAddAtLevel()` below - it adds a new entry.
Good catch - thanks!

Reply:
Done